### DISCUSSION TITLE
How does training iterate over training date? Sequentially? Randomly? How does dataset repetition work?

### DISCUSSION THREAD

**Discussion by reilnuud** [bmaltais/kohya_ss#3003] [Q&A]
I've tried to figure this one out from documentation and other issues but haven't had any luck so far, but how exactly does this library work through training data, especially re: dataset repeats?

Does it move through training data sequentially or "randomly"? If dataset repeats is set, how does it repeat the dataset?

Assuming it's sequential, I'd imagine dataset repeats = 2 works like this:
1,2,3,4,5,6,7,8,9,10,1,2,3,4,5,6,7,8,9,10

Or does it work like so:
1,1,2,2,3,3,4,4,5,5,6,6,7,7,8,8,9,9,10,10

Or, if I'm thinking about this all wrong, I'd love to know how I should be thinking about this.

**Comment by reilnuud** [comment]
Follow up: if I stop and resume from a saved state, does the resumed state start at the same point in the dataset or does it start over?

**Comment by 5KilosOfCheese** [comment]
The order of the images is based on the seed. Which also sets seed for generation the noise and noise amount for the images.
![kuva](https://github.com/user-attachments/assets/6ac69588-b8ca-4bab-8018-cff69f4779f9)

If you really need to figure out the order, you can just use dramatically different sizes of images, then see the amount of time each step takes. You can then figure out the exact order - if you for some reason want to.
